tensorboard/plugins/debugger_v2/debug_data_multiplexer.py

```diff
@@ -98,6 +135,57 @@ def __init__(self, logdir):
         """
         self._logdir = logdir
         self._reader = None
+        self._reader_lock = threading.Lock()
+        self._createOrReloadReader()
+
+    def _createOrReloadReader(self):
+        """Creates or reloads reader for tfdbg2 data in the logdir.
+
+        If the reader has already been created, a new one will not be created;
+        instead, the existing reader's `update()` method will be unblocked on a
+        separate thread immediately.
+
+        If a reader has not been created, create it and start periodic calls to
+        `update()` on a separate thread.
+        """
+        with self._reader_lock:
+            if not self._reader:
+                try:
+                    from tensorflow.python.debug.lib import debug_events_reader
```

Comment:
FWIW, I realize this is existing code, but would be nice if we can avoid doing conditional imports like this and instead use the plugin loader functionality to gate the loading of the entire plugin so that if we can't import the necessary deps we don't end up failing later on like this.

Reply:
I'd like to defer that to a later PR, ideally after tf 2.2 is released so we can remove one of the `except` branches below. Added a TODO item here.